### ISSUE TITLE
Relation Constraints

### ISSUE THREAD

**Issue by ColdAnkles** [zadam/trilium#1991]
When relating one note to another, you get a list of all possible notes that can be related.

If possible, add a statement for relation definitions constraints; for example:

#relation:isChildOf="promoted,inverse=hasChild,multi,constraints=person"

Attempting to fill out any isChildOf relation would only permit relating notes with the #person label.

**Comment by geek-merlin** [comment]
Yes, yes, and yes. Trilium is a powerful low-code database, the biggest lack of which imho is: Enumerations. 

This would add enumerations in a very elegant way.

We can (and should) bikeshed how the constraint is specified, but any implementation is better that not having this.

**Comment by meichthys** [comment]
[Trilium has entered maintenance mode](https://github.com/zadam/trilium/issues/4620#issuecomment-2094495698). Future enhancements will be addressed in TrilumNext: https://github.com/TriliumNext/trilium/issues/5013
